File: GuessingGame/MVGuessingGame.py
# Marlen Vining
# 1/11/22
# Lab #1
#
#This program will do the following:
#Play a simple number guessing game

from random import randint #import random integer

class GuessingGame: #define main class
    def __init__(self, max): #setup function

        #define game variables
        self.max = max

        self.games = 0
        self.guesses = 0

        self.intro() # display intro message

        while True: # main loop
            play = input("Would you like to play a game? (Y/N)?\n") #prompt user to play a game

            if play[0].upper() == "Y": # if yes play game
                self.game()
            elif play[0].upper() == "N": # if no quit
                break
            else:
                print("please input a valid answer") # if neither, tell the user

    def intro(self): #frunction to print an intro message
        print("This program allows you to play a guessing game.")
        print("I will think of a number between 1 and")
        print(f"{max} and will allow you to guess until")
        print("you get it. For each guess, I will tell you")
        print("whether the right answer is higher or lower")
        print("than your guess.")


    def game(self): #function to play a game
        print(f"I'm thinking of a number between 1 and {self.max}") #start message
        randVal = randint(1, self.max) #generate a random number

        self.games += 1 #increment total games
        gameGuesses = 0
        running = True
        while running: #while running play game
            running = self.guess(randVal) #get a guess from user
            gameGuesses += 1

        print(f"game completed in {gameGuesses} guesses") # print stats
        print(f"you take an average of {int(self.guesses / self.games)} to win")

    def guess(self, val): #function to process a guess
        guess = -1 #define guess
        self.guesses += 1 # increment guesses
        while guess == -1: #loop until valid guess
            try:
                guess = int(input("guess a number:\n")) # try to get an int from input
            except:
                print("not a number") # if not, tell the user

            if guess > self.max or guess < 1:
                print(f"please input a number between 1 and {self.max}") #if too high or low, tell the user
                guess = -1

        if guess > val:
            print("guess too high") #if the guess is above the random value, tell the user

        if guess < val: 
            print("guess too low") #if the guess is below, tell the user

        if guess == val: # if the guess is correct, return false to stop the game loop
            return False

        else: #if incorrect, return true to continue the game loop
            return True

if __name__ == "__main__":
    game = GuessingGame(100) # if main, start a game with max value of 100

File: GuessingGame/MVGuessingGame.java
// Marlen Vining
// 1/11/22
// Lab #1
//
//This program will do the following:
//Play a simple number guessing game
//

import java.util.Scanner; //Import scanner class for user input

public class MVGuessingGame {
	private static Scanner scanner = new Scanner(System.in); //creates a scanner to read user input
	
	//set variables for the game
	private static int max = 100;
	private static double games = 0.0;
	private static double guesses = 0.0;
	
	public static void main(String[] args) {
		intro(); //display the intro message
		while (true) { //main loop
			char cmd = input("would you like to play a game? (Y/N)").charAt(0); //ask if the user wants to play a game
			cmd = Character.toUpperCase(cmd); //process the first character
			if (Character.compare(cmd, 'N') == 0) {
				break; //if N, exit the game
			} else if (Character.compare(cmd, 'Y') == 0) {
				games += 1; //if Y, play a game
				game();
			} else {
				System.out.println("Please input a valid answer"); //if neither, prompt the user
			}
		}
	}
	
	static void game() { //method to play a game
		System.out.printf("I'm thinking of a number between 1 and %d\n", max); //print game message
		int rand = (int)(Math.random() * max) + 1; //generate random number
		int roundGuesses = 0;
		
		boolean running = true; //start the game loop
		while (running) {
			int guessInt = -1; //define the guess
			while (guessInt == -1) {//while the guess is not set
				String guess = input("guess a number"); //prompt the user to guess
				try {
					guessInt = Integer.valueOf(guess); //get the int value of guess
					
					if (guessInt > max || guessInt < 0) {
						guessInt = -1; //if too high or low , set to -1
					}
				} catch (Exception e) {
					guessInt = -1; //if error, set to -1
				}
				if (guessInt == -1) {
					System.out.println("Please input a valid number"); //if -1, something must be wrong, prompt the user
				}
			}
			guesses += 1; //increment guesses and round guesses
			roundGuesses += 1;
			int checkVal = check(guessInt, rand); //check the value
			switch (checkVal) { 
				case -1:
					System.out.println("Guess too low"); //if too low, tell the user
					break;
					
				case 1:
					System.out.println("Guess too high"); //if too high, tell the user
					break;
					
				case 0:
					System.out.println("You guessed the number"); //if equal, tell the user and exit to main menu
					System.out.printf("You got it in %d guesses\n", roundGuesses);
					running = false;
					break;
			}
		}
		stats();
	}
	
	static void stats() { //method to print stats
		if (games != 0) {
			System.out.printf("You take an average of %d guesses to win\n", (int)(guesses / games)); //print average guesses per game
		} else {
			System.out.println("Please play at least one game to see your average score"); //don't divide by 0
		}
	}
	
	static int check(int input, int value) { //method to check a guess
		if (input == value) { //return value based on guess and number
			return 0;
		} else if (input > value) {
			return 1;
		} else {
			return -1;
		}
	}
	
	static void intro() { //method to print intro message
		System.out.println("This program allows you to play a guessing game.");
		System.out.println("I will think of a number between 1 and");
		System.out.printf("%d and will allow you to guess until\n", max);
		System.out.println("you get it. For each guess, I will tell you");
		System.out.println("whether the right answer is higher or lower");
		System.out.println("than your guess.");
	}
	
	static String input(String message) { //create an easy way to get user input
		System.out.println(message); //print prompt
		return scanner.next(); // return user input
	}
}